EIPS/eip-6224.md

```diff
@@ -40,119 +40,137 @@ The `ContractsRegistry` MUST implement the following interface:
 pragma solidity ^0.8.0;
 
 interface IContractsRegistry {
-   /**
-    *  @notice REQUIRED The event that is emitted when the contract gets added to the registry
-    *  @param name the name of the contract
-    *  @param contractAddress the address of the added contract
-    *  @param isProxy whether the added contract is a proxy
-    */
-   event AddedContract(string name, address contractAddress, bool isProxy);
-
-   /**
-    *  @notice REQUIRED The event that is emitted when the contract get removed from the registry
-    *  @param name the name of the removed contract
-    */
-   event RemovedContract(string name);
-
-   /**
-    *  @notice REQUIRED The function that returns an associated contract by the name
-    *  @param name the name of the contract
-    *  @return the address of the contract
-    */
-   function getContract(string memory name) external view returns (address);
-
-   /**
-    *  @notice OPTIONAL The function that checks if a contract with a given name has been added
-    *  @param name the name of the contract
-    *  @return true if the contract is present in the registry
-    */
-   function hasContract(string memory name) external view returns (bool);
-
-   /**
-    *  @notice RECOMMENDED The function that returns the admin of the added proxy contracts
-    *  @return the proxy admin address
-    */
-   function getProxyUpgrader() external view returns (address);
-
-   /**
-    *  @notice RECOMMENDED The function that returns an implementation of the given proxy contract
-    *  @param name the name of the contract
-    *  @return the implementation address
-    */
-   function getImplementation(string memory name) external view returns (address);
-
-   /**
-    *  @notice REQUIRED The function that injects dependencies into the given contract.
-    *  MUST call the setDependencies() with address(this) and bytes("") as arguments on the substituted contract
-    *  @param name the name of the contract
-    */
-   function injectDependencies(string memory name) external;
-
-   /**
-    *  @notice REQUIRED The function that injects dependencies into the given contract with extra data.
-    *  MUST call the setDependencies() with address(this) and given data as arguments on the substituted contract
-    *  @param name the name of the contract
-    *  @param data the extra context data
-    */
-   function injectDependenciesWithData(
+    /**
+     *  @notice REQUIRED The event that is emitted when the non-proxy contract gets added to the registry
+     *  @param name the name of the contract
+     *  @param contractAddress the address of the added contract
+     */
+    event ContractAdded(string name, address contractAddress);
```

Comment:
I think these should be indexed.

Reply:
This makes sense, however there might be a cath with long strings that do not fit into 32 bytes index and will be hashed.

Reply:
Fair